sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -201,11 +201,14 @@ case class FilterExec(condition: Expression, child: SparkPlan)
       ev
     }
 
+    // Note: wrap in "do { } while(false);", so the generated checks can jump out with "continue;"
```

Comment:
this is tricky, how hard it is to fix all places that use `continue`?

Reply:
ah i see, you are trying to avoid generating deeply nested if-else branches.

Reply:
`genPredicate` and `generated` ~50 lines above would have to be rewritten to now use `continue`. As you pointed in a previous comment, that would potentially lead to very nested scopes. Shouldn't be a problem for the compiler; for code generation the `genPredicate` would have to maintain these scopes and where to end them - i.e. wherever it not places a `continue`, it would have to open a nested scope, and then it would have to be closed in a correct place.